packages/block-library/src/column/edit.js

```diff
@@ -77,10 +78,18 @@ function ColumnEdit( {
 export default compose(
 	withSelect( ( select, ownProps ) => {
 		const { clientId } = ownProps;
-		const { getBlockOrder } = select( 'core/block-editor' );
+		const {
+			getBlockOrder,
+			getBlockRootClientId,
+			getTemplateLock,
+		} = select( 'core/block-editor' );
+
+		// Retrieve the parent of the Columns Block.
+		const rootParentClientId = getBlockRootClientId( getBlockRootClientId( clientId ) );
```

Comment:
I like this solution, I agree that ideally, the column would inherit the locking from the parent of columns.
I wonder if we can make our code more explicit e.g:
```
const parentColumnsId = getBlockRootClientId( clientId );
const parentOfColumnsId = getBlockRootClientId( parentColumnsId );
```